pom.xml

```
@@ -0,0 +1,378 @@
+    </plugin>
+    <plugin>
+      <groupId>org.jacoco</groupId>
+      <artifactId>jacoco-maven-plugin</artifactId>
+      <reportSets>
+        <reportSet>
+          <reports>
+            <report>report</report>
+          </reports>
+        </reportSet>
+      </reportSets>
+    </plugin>
+  </plugins>
+</reporting>
+
+<profiles>
+  <profile>
+    <id>minimal</id>
+    <activation>
+      <property>
+        <name>containerjfr.minimal</name>
+        <value>true</value>
+      </property>
+    </activation>
+    <properties>
+      <containerjfr.imageVersion>${project.version}-minimal</containerjfr.imageVersion>
+    </properties>
+  </profile>
+  <profile>
+    <id>non-minimal</id>
+    <activation>
+      <property>
+        <name>containerjfr.minimal</name>
+        <value>!true</value>
+      </property>
+    </activation>
+    <properties>
+      <containerjfr.imageVersion>${project.version}</containerjfr.imageVersion>
+    </properties>
+    <build>
+      <plugins>
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>exec-maven-plugin</artifactId>
+          <version>1.6.0</version>
+          <configuration>
+            <mainClass>com.redhat.rhjmc.containerjfr.ContainerJfr</mainClass>
+          </configuration>
+          <executions>
+            <execution>
+              <id>update-web-client-deps</id>
+              <phase>initialize</phase>
+              <goals>
+                <goal>exec</goal>
+              </goals>
+              <configuration>
+                <executable>npm</executable>
+                <workingDirectory>web-client</workingDirectory>
+                <arguments>
+                  <argument>ci</argument>
+                </arguments>
+              </configuration>
+            </execution>
+            <execution>
+              <id>build-web-client</id>
+              <phase>prepare-package</phase>
+              <goals>
+                <goal>exec</goal>
+              </goals>
+              <configuration>
+                <executable>npm</executable>
+                <workingDirectory>web-client</workingDirectory>
+                <arguments>
+                  <argument>run</argument>
+                  <argument>build</argument>
+                </arguments>
+              </configuration>
+            </execution>
+            <execution>
+              <id>clean-web-client</id>
+              <phase>clean</phase>
+              <goals>
+                <goal>exec</goal>
+              </goals>
+              <configuration>
+                <executable>rm</executable>
+                <workingDirectory>web-client</workingDirectory>
+                <arguments>
+                  <argument>-rf</argument>
+                  <argument>node_modules</argument>
+                  <argument>dist</argument>
+                </arguments>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
```

Might I suggest we handle clean using maven-clean-plugin? Seems a bit more portable.

```suggestion
          </executions>
        </plugin>
        <plugin>
          <artifactId>maven-clean-plugin</artifactId>
          <version>3.1.0</version>
          <configuration>
            <filesets>
              <fileset>
                <directory>web-client</directory>
                <includes>
                  <include>node_modules/**</include>
                  <include>dist/**</include>
                </includes>
              </fileset>
            </filesets>
          </configuration>
        </plugin>
```